packages/expo-image-manipulator/ios/ImageManipulations.swift

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021-present 650 Industries. All rights reserved.
+
+import UIKit
+
+/**
+ Main `manipulate` function that takes an array of any supported actions to apply.
+ */
+internal func manipulate(image initialImage: UIImage, actions: [ManipulateAction]) throws -> UIImage {
+  var image: UIImage = initialImage
+
+  image = try fixImageOrientation(image)
+
+  for action in actions {
+    if let resize = action.resize {
+      image = try manipulate(image: image, resize: resize)
+    } else if let rotate = action.rotate {
+      image = try manipulate(image: image, rotate: rotate)
+    } else if let flip = action.flip {
+      image = try manipulate(image: image, flip: flip)
+    } else if let crop = action.crop {
+      image = try manipulate(image: image, crop: crop)
+    }
```

Comment:
Pattern matching is not an alternative for boolean conditions nor let assignments, so I'm not sure what do you mean here 🤔

Reply:
I imagined it might be possible to use such `switch case let` construction (I've seen something like this in this article: https://useyourloaf.com/blog/swift-if-case-let/):
```swift
try image = switch (action) {
  case let resize = action.resize: manipulate(image: image, resize: resize),
  case let rotate = action.rotate: manipulate(image: image, rotate: rotate),
  ...
}
```

Reply:
Yeah, that's what I guessed but generally pattern matching works only with enums, so it doesn't apply here 😅